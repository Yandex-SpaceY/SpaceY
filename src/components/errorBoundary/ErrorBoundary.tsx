import React, { Component, ErrorInfo, ReactNode } from 'react';

import { Oops } from 'components';

type ErrorState = { isError: boolean }

class ErrorBoundary extends Component {
  state: ErrorState = {
    isError: false,
  }

  static getDerivedStateFromError(): ErrorState {
    return { isError: true };
  }

  componentDidCatch(error: Error, errorInfo: ErrorInfo): void {
    console.error(`Pay attention to the error. ${error}: ${errorInfo}`);
  }

  clearState = (): void => {
    this.setState({
      isError: false,
    });
  }

  render(): ReactNode {
    return this.state.isError ? <Oops clearState={this.clearState} /> : this.props.children;
  }
}

export default ErrorBoundary;
