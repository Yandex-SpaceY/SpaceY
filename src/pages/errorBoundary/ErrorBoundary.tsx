import React, { Component } from 'react';

import OopsPage from './components/oopsPage/OopsPage';

type ErrorState = { error: Error | null, errorInfo: React.ErrorInfo | null}

class ErrorBoundary extends Component {
  state: ErrorState = {
    error: null,
    errorInfo: null,
  }

  componentDidCatch(error: Error, errorInfo: React.ErrorInfo): void {
    this.setState({
      error,
      errorInfo,
    });
  }

  render(): React.ReactNode {
    return this.state.errorInfo ? <OopsPage /> : this.props.children;
  }
}

export default ErrorBoundary;
