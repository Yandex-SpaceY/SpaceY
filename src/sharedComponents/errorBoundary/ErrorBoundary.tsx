import React, { Component } from 'react';
import { withRouter, RouteComponentProps } from 'react-router-dom';

import { routeConstants } from 'constants/routeConstants';
import { Oops } from 'pages/';

type ErrorState = { error: Error | null, errorInfo: React.ErrorInfo | null}

class ErrorBoundary extends Component<RouteComponentProps, ErrorState> {
  state = {
    error: null,
    errorInfo: null,
  }

  componentDidCatch(error: Error, errorInfo: React.ErrorInfo): void {
    this.props.history.push(routeConstants.OOPS);

    this.setState({
      error,
      errorInfo,
    });
  }

  render(): React.ReactNode {
    return this.state.errorInfo ? <Oops /> : this.props.children;
  }
}

export default withRouter(ErrorBoundary);
