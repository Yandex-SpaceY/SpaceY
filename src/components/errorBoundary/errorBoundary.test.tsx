import * as React from 'react';
import { shallow } from 'enzyme';

import ErrorBoundary from './ErrorBoundary';

it('check render component ErrorBoundary', () => {
  const testText = 'no error children text';
  const component = shallow(<ErrorBoundary>{testText}</ErrorBoundary>);

  expect(component.text()).toEqual(testText);

  expect(component).toMatchSnapshot();
});
