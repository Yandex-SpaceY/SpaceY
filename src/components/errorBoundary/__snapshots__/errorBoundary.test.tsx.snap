// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check render component ErrorBoundary 1`] = `"no error children text"`;
